src/FastIntegerDivide.cpp

```diff
@@ -205,6 +245,46 @@ Expr fast_integer_divide(Expr numerator, Expr denominator) {
 
         // Maybe flip the bits again
         result = xsign ^ result;
+    } else {
+        // Signed round to zero
+        Expr mul, shift = shift_for_denominator(denominator);
+        switch (t.bits()) {
+        case 8: {
+            Buffer<uint8_t> table = integer_divide_table_srz8();
+            mul = table(denominator);
+            break;
+        }
+        case 16: {
+            Buffer<uint16_t> table = integer_divide_table_srz16();
+            mul = table(denominator);
+            break;
+        }
+        default:  // 32
+        {
+            Buffer<uint32_t> table = integer_divide_table_srz32();
+            mul = table(denominator);
+            break;
+        }
+        }
+
+        // Extract sign bit
+        // Expr xsign = (t.bits() < 32) ? (numerator / (1 << (t.bits()-1))) : (numerator >> (t.bits()-1));
+        Expr xsign = select(numerator > 0, cast(t, 0), cast(t, -1));
+
+        // Multiply-keep-high-half
+        result = (cast(wide, mul) * numerator);
```

Comment:
I think this should use `widening_mul` intrinsics, because uses of this are after `find_intrinsics`. Maybe this whole sequence should be `mul_shift_right`.

Reply:
This is just moving around the original code, but sure I can update it.

Reply:
Actually this code is only called directly by users, so it's before find_intrinsics. The compiler doesn't ever call this.

Reply:
Maybe add this as a comment for future readers.

Reply:
I actually think we should change it to intrinsics anyways. But since the code is just moved and pre-existing, maybe it should be a separate PR.